wzdx/raw_to_standard/icone.py

```diff
@@ -0,0 +1,223 @@
+        return None
+    # polyline rightnow is a list which has an empty string in it.
+    polyline = polylinestring.split(',')
+    coordinates = []
+    for i in range(0, len(polyline)-1, 2):
+        try:
+            coordinates.append([float(polyline[i + 1]), float(polyline[i])])
+        except ValueError as e:
+            logging.warning('failed to parse polyline!')
+            return []
+    return coordinates
+
+
+def get_sensor_list(incident):
+    devices = []
+    for key in ['sensor', 'radar', 'display', 'message', 'marker', 'status']:
+        obj = incident.get(f"{key}")
+        print("-------------------")
+        print(key)
+        print(obj)
+        print(type(obj))
+        if type(obj) in [dict, OrderedDict]:
+            devices.append({'sensor_type': key, 'details': obj})
+        elif type(obj) == list:
+            for i in obj:
+                devices.append({'sensor_type': key, 'details': i})
+    return devices
+
+
+def create_rtdh_standard_msg(pd):
+    devices = get_sensor_list(pd.get(f"incident"))
+    return {
+        "rtdh_timestamp": time.time(),
+        "rtdh_message_id": str(uuid.uuid4()),
+        "event": {
+            "type": pd.get("incident/type", default=""),
+            "source": {
+                "id": pd.get("incident/@id", default=""),
+                "creation_timestamp": pd.get("incident/creationtime", date_tools.get_unix_from_iso_string, default=0),
+                "last_updated_timestamp": pd.get("incident/updatetime", date_tools.get_unix_from_iso_string, default=0),
+            },
+            "geometry": pd.get("incident/location/polyline", parse_icone_polyline),
+            "header": {
+                "description": pd.get("incident/description", default=""),
+                "start_timestamp": pd.get("incident/starttime", date_tools.get_unix_from_iso_string, default=None),
+                "end_timestamp": pd.get("incident/endtime", date_tools.get_unix_from_iso_string, default=None)
+            },
+            "detail": {
+                "road_name": pd.get("incident/location/street"),
+                "road_number": pd.get("incident/location/street"),
+                "direction": get_direction(pd.get("incident/location/street"), pd.get("incident/location/polyline"))
+            },
+            "additional_info": {
+                "devices": devices,
+                "directionality": pd.get("incident/location/direction")
+            }
+        }
+    }
+
+
+def get_direction(street, coords):
+    direction = wzdx_translator.parse_direction_from_street_name(street)
+    if not direction:
+        direction = polygon_tools.get_road_direction_from_coordinates(coords)
+    return direction
+
+
+# function to parse polyline to geometry line string
+def parse_icone_polyline(polylinestring):
+    if not polylinestring or type(polylinestring) != str:
+        return None
+    # polyline rightnow is a list which has an empty string in it.
```

Comment:
This method seems to be duplicated from line 86. Can we remove?

Reply:
good catch, removed